curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/660f2b6fd54ac1fc142804dd.md

```diff
@@ -0,0 +1,57 @@
+---
+id: 660f2b6fd54ac1fc142804dd
+title: Step 37
+challengeType: 0
+dashedName: step-37
+---
+
+# --description--
+
+The logic for formatting this pyramid is likely going to get complicated, which means it's a great time to extract that code into a function.
+
+A <dfn>function</dfn> is a block of code that can be reused throughout your application. Functions are declared with the following syntax:
```

Comment:
I think this explanation is good on functions, but it is missing three keys things

1. undefined values from functions
2. global scope and functions
3. block scope and functions

Looking at the legacy curriculum, we should have this block of lessons 
<img width="762" alt="Screenshot 2024-04-05 at 5 12 48 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/385e2297-0091-4e40-a4d9-caa035c9d736">

some the examples in that block are a little long. so we can use different examples. 
but that block does cover everything we need to create a strong foundation on how functions work.

Plus it will clear up a lot of the confusion we are seeing on the forum and discord. 

I think it would work better to have a set of steps just for functions with some sort of dummy example they can walk through. 

then they can delete the code they wrote from that example and apply what they have learned to the pyramid.